cpp/include/cudf/hashing/detail/xxhash_64.cuh

```diff
@@ -29,7 +29,8 @@ namespace cudf::hashing::detail {
 
 template <typename Key>
 struct XXHash_64 {
-  using result_type = std::uint64_t;
+  using argument_type = Key;
```

Comment:
Why is this needed? Do we not know the type that the hasher is templated with?

The member type `std::hash<T>::argument_type` is deprecated in C++17 and removed in C++20, so we should probably take a close look at what the STL is doing here. https://en.cppreference.com/w/cpp/utility/hash

Reply:
Agreed. We know the argument type aka the `Key`, but for now we need it to properly instantiate cuco::default/arrow_filter_policy here: https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/bloom_filter/arrow_filter_policy.cuh#L89
https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/bloom_filter/default_filter_policy_impl.cuh#L36

Let me create a cuco issue to remove this.

Reply:
Related cuco issue: https://github.com/NVIDIA/cuCollections/issues/653

Reply:
We are aware of this change. It's just so convenient to have these types added.

Reply:
I’m okay with that! I just didn’t want us to add a member type if it was going to be removed soon. 😄

Reply:
This is a way to handle this. Let me explain in the corresponding cuco issue.

Reply:
https://github.com/NVIDIA/cuCollections/pull/654 reverts the need for `argument_type` so reverting.